libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/HttpRequest.cs

```diff
@@ -228,187 +228,196 @@ public override async Task<DialogTurnResult> BeginDialogAsync(DialogContext dc,
-                    if (instanceBody == null)
-                    {
-                        using (var request = new HttpRequestMessage(new System.Net.Http.HttpMethod("PATCH"), instanceUrl))
+                    case HttpMethod.PATCH:
+                        if (instanceBody == null)
                         {
                             response = await client.SendAsync(request, cancellationToken).ConfigureAwait(false);
                         }
-                    }
-                    else
-                    {
-                        using (var request = new HttpRequestMessage(new System.Net.Http.HttpMethod("PATCH"), instanceUrl))
+                        else
                         {
                             request.Content = new StringContent(instanceBody.ToString(), Encoding.UTF8, contentType);
                             traceInfo.request.content = instanceBody.ToString();
                             traceInfo.request.headers = JObject.FromObject(request.Content.Headers.ToDictionary(t => t.Key, t => (object)t.Value?.FirstOrDefault()));
                             response = await client.SendAsync(request, cancellationToken).ConfigureAwait(false);
                         }
-                    }
 
-                    break;
+                        break;
 
-                case HttpMethod.PUT:
-                    if (instanceBody == null)
-                    {
-                        response = await client.PutAsync(instanceUrl, content: null, cancellationToken: cancellationToken).ConfigureAwait(false);
-                    }
-                    else
-                    {
-                        using (var putContent = new StringContent(instanceBody.ToString(), Encoding.UTF8, contentType))
+                    case HttpMethod.PUT:
+                        if (instanceBody == null)
                         {
-                            traceInfo.request.content = instanceBody.ToString();
-                            traceInfo.request.headers = JObject.FromObject(putContent.Headers.ToDictionary(t => t.Key, t => (object)t.Value?.FirstOrDefault()));
-                            response = await client.PutAsync(instanceUrl, putContent, cancellationToken).ConfigureAwait(false);
+                            response = await client.SendAsync(request, cancellationToken).ConfigureAwait(false);
+                        }
+                        else
+                        {
+                            using (var putContent = new StringContent(instanceBody.ToString(), Encoding.UTF8, contentType))
+                            {
+                                traceInfo.request.content = instanceBody.ToString();
+                                traceInfo.request.headers = JObject.FromObject(putContent.Headers.ToDictionary(t => t.Key, t => (object)t.Value?.FirstOrDefault()));
+                                request.Content = new StringContent(instanceBody.ToString(), Encoding.UTF8, contentType);
```

Comment:
Isn't this the putContent we created in the using in line 332? Shouldn't we be using that disposable instance instead of creating a new instance of StringContent?